instrumentation/grails-3.0/javaagent/build.gradle.kts

```diff
@@ -43,8 +43,4 @@ dependencies {
 
   testLibrary("org.springframework.boot:spring-boot-autoconfigure:$springBootVersion")
   testLibrary("org.springframework.boot:spring-boot-starter-tomcat:$springBootVersion")
-
-  // spring boot 3 uses spring 6 which requires java 17
-  latestDepTestLibrary("org.springframework.boot:spring-boot-autoconfigure:2.+")
```

Comment:
This will still be a problem when Spring 3 is released

Reply:
is your preference to keep this and open a tracking issue now?

Reply:
I think yes, that will remove a time pressure from us.

Reply:
by "remove time pressure" do you mean it will give us more time to fix the issue and support spring boot 3 before it's GA'd? if this is a concern, maybe we should add some exceptions to #5258?

Reply:
No, I mean that if we keep this constraint, then when Spring Boot 3 is released we will NOT be forced to fix it ASAP because our build broke.

Reply:
I personally think that's a good thing, it's sort of an alarm bell that Spring Boot 3 has GA'd. It's easy enough to snooze the alarm by adding the suppression + logging the issue at that time.

Reply:
going to merge, I think removing it is most consistent from #5258 perspective, and I think the alarm bell when unsupported GAs are released is a good thing for us, but we can revisit if this feels problematic